src/Workspaces/Remote/ServiceHub/Services/AssetSynchronization/RemoteAssetSynchronizationService.cs

```diff
@@ -3,112 +3,116 @@
-    /// an edit once a feature is asking for a snapshot.
-    /// </summary>
-    internal sealed class RemoteAssetSynchronizationService : BrokeredServiceBase, IRemoteAssetSynchronizationService
+    internal sealed class Factory : FactoryBase<IRemoteAssetSynchronizationService>
     {
-        internal sealed class Factory : FactoryBase<IRemoteAssetSynchronizationService>
-        {
-            protected override IRemoteAssetSynchronizationService CreateService(in ServiceConstructionArguments arguments)
-                => new RemoteAssetSynchronizationService(in arguments);
-        }
-
-        public RemoteAssetSynchronizationService(in ServiceConstructionArguments arguments)
-            : base(in arguments)
-        {
-        }
+        protected override IRemoteAssetSynchronizationService CreateService(in ServiceConstructionArguments arguments)
+            => new RemoteAssetSynchronizationService(in arguments);
+    }
 
-        public ValueTask SynchronizePrimaryWorkspaceAsync(Checksum solutionChecksum, CancellationToken cancellationToken)
+    public ValueTask SynchronizePrimaryWorkspaceAsync(Checksum solutionChecksum, CancellationToken cancellationToken)
+    {
+        return RunServiceAsync(async cancellationToken =>
         {
-            return RunServiceAsync(async cancellationToken =>
+            using (RoslynLogger.LogBlock(FunctionId.RemoteHostService_SynchronizePrimaryWorkspaceAsync, Checksum.GetChecksumLogInfo, solutionChecksum, cancellationToken))
             {
-                using (RoslynLogger.LogBlock(FunctionId.RemoteHostService_SynchronizePrimaryWorkspaceAsync, Checksum.GetChecksumLogInfo, solutionChecksum, cancellationToken))
-                {
-                    var workspace = GetWorkspace();
-                    var assetProvider = workspace.CreateAssetProvider(solutionChecksum, WorkspaceManager.SolutionAssetCache, SolutionAssetSource);
-                    await workspace.UpdatePrimaryBranchSolutionAsync(assetProvider, solutionChecksum, cancellationToken).ConfigureAwait(false);
-                }
-            }, cancellationToken);
-        }
+                var workspace = GetWorkspace();
+                var assetProvider = workspace.CreateAssetProvider(solutionChecksum, WorkspaceManager.SolutionAssetCache, SolutionAssetSource);
+                await workspace.UpdatePrimaryBranchSolutionAsync(assetProvider, solutionChecksum, cancellationToken).ConfigureAwait(false);
+            }
+        }, cancellationToken);
+    }
+
+    public ValueTask SynchronizeActiveDocumentAsync(DocumentId? documentId, CancellationToken cancellationToken)
+    {
+        var documentTrackingService = GetWorkspace().Services.GetRequiredService<IDocumentTrackingService>() as RemoteDocumentTrackingService;
+        documentTrackingService?.SetActiveDocument(documentId);
+        return ValueTaskFactory.CompletedTask;
```

Comment:
impl on OOP side is trivial.  we literally just set the value in the oop doc tracking service.